python/ql/src/Security/CWE-602/DangerousHttpServer.ql

```diff
@@ -0,0 +1,26 @@
+/**
+ * @name Dangerous http.server module
+ * @description Use of a module that is not recommended for production, as it only implements basic security checks.
+ * @kind problem
+ * @problem.severity warning
+ * @precision high
+ * @id python/dangerous-http-server
+ * @tags reliability
+ *       security
+ */
+
+
+import python
+
+ModuleValue http_server(string mod, string msg) {
+    mod = "http.server"
+    and
+    msg = "http.server is not recommended for production. It only implements basic security checks."
+    and
+    result = Module::named(mod)
+}
+
+from AstNode c, string mod, string msg
+where
+    c = http_server(mod, msg).getAReference().getNode()
+select c, msg
```

Comment:
Why not simplify this?
```suggestion
from AstNode c
where
    c = Module::named("http.server").getAReference().getNode()
select c, "http.server is not recommended for production. It only implements basic security checks."
```

Reply:
You're right! I was originally going to add more to this query which would have required the structure I used originally. But this is much more concise.